cmd/lakectl/cmd/root.go

```diff
@@ -428,8 +439,14 @@ func getClient() *apigen.ClientWithResponses {
 	// see: https://stackoverflow.com/a/39834253
 	transport := http.DefaultTransport.(*http.Transport).Clone()
 	transport.MaxIdleConnsPerHost = DefaultMaxIdleConnsPerHost
-	httpClient := &http.Client{
-		Transport: transport,
+	var httpClient *http.Client
+	if !cfg.Server.Retries.Enabled {
+		httpClient = &http.Client{
+			Transport: transport,
+		}
+	} else {
+		httpClient = NewRetryClient(cfg.Server.Retries.MaxRetries,
+			cfg.Server.Retries.MinWaitInterval, cfg.Server.Retries.MaxWaitInterval, transport)
```

Comment:
nit: but I would expect the newRetryClient to get the httpClient and not the transport, that we can initiate the client regardless and then just wrap it with the retry client if needed